scalding-core/src/test/scala/com/twitter/scalding/ReferencedClassFinderTest.scala

```diff
@@ -0,0 +1,49 @@
+package com.twitter.scalding
+
+import org.apache.hadoop.io.BytesWritable
+import org.scalatest.{ Matchers, WordSpec }
+
+case class C1(a: Int)
+case class C2(b: Int)
+case class C3(c: Int)
+case class C4(d: Int)
+
+trait TraitType {
```

Comment:
do we need to add some tests for:
1) Java types? 
2) Thrift / protobuf types?
3) Testing that if we have some primitive types (like Int) referenced, we don't end up adding their tokens? (Or is that the same scenario as the BytesWriteable?)